tests/testapp/tests.py

```diff
@@ -79,17 +82,105 @@ def authenticate(user, filter_kwargs):
         run_scenarios_with(authenticate)
 
 
-class ActAsModelBackendTestCase(TransactionTestCase):
+class TestableBackend(object):
 
-    def test_it_is_a_filtered_model_backend(self):
-        self.assertTrue(
-            issubclass(ActAsModelBackend, FilteredModelBackend),
-            ActAsModelBackend.__mro__)
+    def __init__(self):
+        self.reset()
+
+    def authenticate(self, *a, **kw):
+        if django_11_or_later:
+            kw.pop('request')
+        self.calls.append((a, kw))
+        return self.authenticated_user
+
+    def reset(self):
+        self.calls = []
+        self.authenticated_user = None
+
+
+def patched_get_backends(backends):
+    method_to_patch = \
+        '_get_backends' if django_11_or_later else 'get_backends'
+    return patch(
+        'django.contrib.auth.{}'.format(method_to_patch),
+        return_value=backends
+    )
+
+
+class ActAsBackendAuthenticateTestCase(TransactionTestCase):
+
+    def setUp(self):
+        super(ActAsBackendAuthenticateTestCase, self).setUp()
+        self.first_test_backend = TestableBackend()
+        self.second_test_backend = TestableBackend()
+        self.third_test_backend_not_in_get_backends = TestableBackend()
+        self.act_as_auth_backend = ActAsBackend()
+        self.backends = [
+            self.first_test_backend,
+            self.act_as_auth_backend,
+            self.second_test_backend
+        ]
 
-    def test_can_authenticate_user(self):
-        user = create_user(username='user', password='password')
+    def patched_get_backends(self):
+        return patched_get_backends(self.backends)
+
+    def test_does_not_inherit_from_any_backend(self):
```

Comment:
I'm unsure about the need of this test, isn't this a try to make it foolproof? Or in other words, test that I wrote `object` and it means `object`?